common/lib/xmodule/xmodule/course_module.py

```diff
@@ -910,6 +918,48 @@ def __init__(self, *args, **kwargs):
             self.discussion_topics = {_('General'): {'id': self.location.html_id()}}
 
         try:
+            enrollment_track_scheme = UserPartition.get_scheme("enrollment_track")
+            enrollment_track_partitions = self.get_user_partitions_for_scheme(enrollment_track_scheme)
+            # Note that when a course_module is initially created, the ID is the draft version.
+            # We just want to store the course_id without published or draft.
+            course_id = unicode(self.id.for_branch(None))
```

Comment:
FYI, I'm working on this refactor to change group from a property to a method on a separate branch, as it impacts a lot of files and I expect I won't get them all right the first time.